### PR TITLE
k8s: Disable several CiliumEndpoint status sections by default

### LINKED ISSUE #10413: Make CEP status optional

### PR DESCRIPTION
**Pull request by tgraf**
The PATCH'ing of the status field of the CiliumEndpoint is often the bottleneck
in scalability. However, most of the frequently changing status sections
provided are not required for operation and only provide additional visibility.

Disable several CiliumEndpoint status sections by default to improve the
scalability of the CRD-backed identity allocation mode. Introduce a flag
--endpoint-status to re-enable individual status sections.

Fixes: #10413

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10490)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by jrajahalme** [comment]
CI tests seem to depend on the CEP status updates?

**Comment by tgraf** [comment]
> CI tests seem to depend on the CEP status updates?

Yes, looking into it.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
Waiting for reviews before rebasing

**Review by nebril — APPROVED** [review]

File: daemon/daemon_main.go
Comment (errordeveloper) [inline daemon/daemon_main.go]: Should this include example values or at least a pointer to what those value can be?

File: pkg/endpoint/endpoint_status_test.go
Comment (errordeveloper) [inline pkg/endpoint/endpoint_status_test.go:52]: Would `map[string]struct{}` be sufficient?

**Review by jrajahalme — APPROVED** [review]

File: pkg/endpoint/endpoint_status_test.go
Comment (tgraf) [reply pkg/endpoint/endpoint_status_test.go:52]: Technically, yes.  I think that the `bool` type self-documents the test better as it makes it obvious that a particular feature is enabled. Given these are unit tests, there is no need to micro-optimize memory use.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: test/k8sT/Services.go
Comment (joestringer) [inline test/k8sT/Services.go:841]: I think that there's a subtle difference here where previously the endpoint plumbing was tested, but now it's only the policy level plumbing that's tested. Arguably we could achieve the exact same as before using `cilium endpoint list -o json`.

That said, I'm not sure whether this particular detail is necessary to validate in these tests so I'll leave it up to your discretion whether this is worth further modifying.

**Review by joestringer — APPROVED** [review]
LGTM, one minor note below.

File: test/k8sT/Services.go
Comment (tgraf) [reply test/k8sT/Services.go:841]: Yes, there is a difference. The rationale here is that we a lot of tests already covering CIDR + label policies to endpoint plumbing combined with connectivity tests. What seemed vital is to test the correct translation of service-based policies to the relevant CIDR policies.

**Comment by tgraf** [comment]
test-me-please

File: Documentation/cmdref/cilium-agent.md
Comment (aanm) [inline Documentation/cmdref/cilium-agent.md]: Why not an "all"?

File: pkg/option/config.go
Comment (aanm) [inline pkg/option/config.go]: Does not hurt to return this by order with `sort.Strings()` or ordering them manually. before returning this slice.

File: Documentation/cmdref/cilium-agent.md
Comment (tgraf) [reply Documentation/cmdref/cilium-agent.md]: Because most users will unlikely understand the scalability consequences of enabling all options. Given the consequences, it seems more sensible to require users to opt-in by feature.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**
